independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -223,7 +231,7 @@ private void fireIfNotEmpty(LazyValue<Notifier<Object>> value) {
         @Override
         public Map<InjectableBean<?>, Object> getContextualInstances() {
             return value.values().stream()
-                    .collect(Collectors.toMap(ContextInstanceHandle::getBean, ContextInstanceHandle::get));
+                    .collect(Collectors.toUnmodifiableMap(ContextInstanceHandle::getBean, ContextInstanceHandle::get));
```

Comment:
Might want to mention that it's unmodifiable in javadoc of `InjectableContext.ContextState#getContextualInstances`

Reply:
It's there: https://github.com/quarkusio/quarkus/pull/22095/files#diff-ce98cefd3872a8ca7303d78b930b1200ea588634d6c868becdc86ba497dfb999R94

Reply:
Ah, missed that, sorry.